tests/type_inference_test.py

```diff
@@ -0,0 +1,193 @@
+import unittest
+import numpy as np
+from dace.config import Config
+from dace.codegen.tools import type_inference
+from dace import dtypes
+import ast
+
+
+class TestTypeInference(unittest.TestCase):
+    def testSimpleAssignement(self):
+        # simple assignment tests
+
+        #bool
+        code_str = "value=True"
+        inf_symbols = type_inference.infer(code_str, {})
+        self.assertEqual(inf_symbols["value"], dtypes.typeclass(bool))
+
+        # int
+        code_str = "value = 1"
+        inf_symbols = type_inference.infer(code_str, {})
+        self.assertEqual(inf_symbols["value"], dtypes.typeclass(int))
+
+        # float
+        code_str = "value = 1.1"
+        inf_symbols = type_inference.infer(code_str, {})
+        self.assertEqual(inf_symbols["value"], dtypes.typeclass(float))
+
+        # assignment with previous symbols
+        prev_symbols = {"char_num": dtypes.typeclass(np.int8)}
+        code_str = "value =  char_num"
+        inf_symbols = type_inference.infer(code_str, prev_symbols)
+        self.assertEqual(inf_symbols["value"], dtypes.typeclass(np.int8))
+
+        # aug assignment
+        code_str = "value += 1.1"
+        inf_symbols = type_inference.infer(code_str, {})
+        self.assertEqual(inf_symbols["value"], dtypes.typeclass(float))
+
+        # annotated assignments
+        # in this case conversion is stricter (int-> int32)
+        code_str = "value : int  = 1"
```

Comment:
how do we know `int` is int32?

Reply:
Because for annotate assignments it uses (defined in `cppunparse._py2c_typeconversion`)
```
_py2c_typeconversion = {
    "uint": dace.dtypes.typeclass(np.uint32),
    "int": dace.dtypes.typeclass(np.int32),
    "float": dace.dtypes.typeclass(np.float32),
    "float64": dace.dtypes.typeclass(np.float64)
}

```

Reply:
should it have a different definition than python? when I write `x: int = 4` I expect the value to be a Python int.